src/ecs/resources/input.rs

```diff
@@ -116,86 +174,65 @@ impl InputHandler {
         }
     }
 
+    /// Returns a string representation of all text entered this frame.
+    ///
+    /// Intended for use with text entry fields, insert this string at the cursor position
+    /// every frame.
+    pub fn text_entered(&self) -> &str {
+        self.text_this_frame.as_str()
+    }
+
     /// Returns an iterator for all the pressed down keys.
     pub fn pressed_keys(&self) -> PressedKeys {
-        PressedKeys { iterator: self.pressed_keys.keys() }
+        PressedKeys { iterator: self.pressed_keys.iter() }
     }
 
     /// Checks if the given key is being pressed.
-    pub fn key_down(&self, key: VirtualKeyCode) -> bool {
-        self.pressed_keys.contains_key(&key)
+    pub fn key_pressed(&self, key: VirtualKeyCode) -> bool {
+        self.pressed_keys.iter().any(|&k| k == key)
     }
 
     /// Checks if all the given keys are being pressed.
-    pub fn keys_down(&self, keys: &[VirtualKeyCode]) -> bool {
+    pub fn keys_pressed(&self, keys: &[VirtualKeyCode]) -> bool {
```

Comment:
`keys_pressed` and `key_pressed` seem to have a different behavior as the first one uses `key_down` internally.